src/committee/orchestrator.rs

```diff
@@ -75,79 +83,120 @@ impl Orchestrator {
         // Round 1
         //
 
-        let mut commitments_map = BTreeMap::new();
+        // Round 1 - Randomly pick some members
+        let results = Arc::new(Mutex::new(vec![]));
+        let mut all_members = self.committee_cfg.members.iter().collect_vec();
 
-        // pick a threshold of members at random
-        // TODO: AT RANDOM!
-        let threshold_of_members = self
-            .committee_cfg
-            .members
-            .iter()
-            .take(self.committee_cfg.threshold)
-            .collect_vec();
+        all_members.shuffle(&mut rand::thread_rng());
+        let mut stream = tokio_stream::iter(&all_members);
+
+        // Sending requests to the selected members concurrently
+        loop {
+            let (member_id, member) = match stream.next().await {
+                Some(item) => item,
+                None => break,
```

Comment:
you can replace this loop with `while let Some((member_id, member)) = stream.next().await {`